storage/notification_cleaner.go

```diff
@@ -64,12 +65,25 @@ func (nc *NotificationCleaner) Start(ctx context.Context, group *sync.WaitGroup)
 }
 
 func (nc *NotificationCleaner) clean(ctx context.Context) {
-	cleanTimestamp := time.Now().Add(-nc.Settings.Notification.OlderThan).Format(time.RFC3339)
+	cleanTimestamp := time.Now().Add(-nc.Settings.Notification.KeepFor).Format(time.RFC3339)
 	log.C(ctx).Infof("Deleting notifications created before %s", cleanTimestamp)
 
 	q := query.ByField(query.LessThanOperator, "created_at", cleanTimestamp)
-	_, err := nc.Storage.Delete(ctx, types.NotificationType, q)
-	if err != nil && err != util.ErrNotFoundInStorage {
+	deletedNotifications, err := nc.Storage.Delete(ctx, types.NotificationType, q)
+
+	if isNotFoundError(err) {
+		log.C(ctx).Debug("no old notifications to delete")
+	} else if err != nil {
 		log.C(ctx).WithError(err).Error("could not delete old notifications")
+	} else {
+		log.C(ctx).Infof("successfully deleted %d old notifications", deletedNotifications.Len())
+	}
+}
+
+func isNotFoundError(err error) bool {
```

Comment:
PR #261 addresses the issue that handleStorageError is called from multiple places. I'd suggest merging master into the `notifications` branch and then reverting this here

Reply:
Will be fixed when merging master into notifications